include/vast/Util/DialectConversion.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022, Trail of Bits, Inc.
+
+#pragma once
+
+VAST_RELAX_WARNINGS
+#include <mlir/Transforms/DialectConversion.h>
+VAST_UNRELAX_WARNINGS
+
+namespace vast::util
+{
+    template< typename Op >
+    struct State
+    {
+        using rewriter_t = mlir::ConversionPatternRewriter;
```

Comment:
I suggest to create some common usings for rewrite patterns at some shared place ... similar to Util/Common.hpp